### PR TITLE
Logging fix

### PR DESCRIPTION
**Pull request by karanjitsingh**
Fix time stamps
Fix missing stdout/stderror
Log output in vstest EqtTrace
Print process id along with start time

### CODE REVIEW

File: src/JSTest/JSProcess.cs [hunk 1]
```diff
@@ -31,6 +31,21 @@ public CommunicationChannel CommunicationChannel
             }
         }
 
+        public int ProcessId
+        {
+            get
+            {
+                try
+                {
+                    return this.process.Id;
```
Comment (cltshivash) [inline src/JSTest/JSProcess.cs]: >process [](start = 32, length = 7)

you could assign it when the process is assigned ..?

File: src/JSTest/JSProcess.cs [hunk 2]
```diff
@@ -45,7 +60,7 @@ public void EnableDebugLogs(string debugFilePath)
             this.debugFilePath = debugFilePath;
         }
 
-        public bool LaunchProcess(TestProcessStartInfo startInfo, Action<object, string> processErrorReceived, Action<object> processExitReceived)
+        public bool LaunchProcess(TestProcessStartInfo startInfo, Action<object, string> processOutputRecieived, Action<object, string> processErrorReceived, Action<object> processExitReceived)
```
Comment (cltshivash) [inline src/JSTest/JSProcess.cs]: >Action<object, string> processOutputRecieived, Actio [](start = 66, length = 52)

nitpick: multiple delegates ... opportunity to move to interface..?

**Comment by cltshivash** [comment]
>                         p.WaitForExit(500);

log the fact that you failed on waitforexit.. 

---
Refers to: src/JSTest/JSProcess.cs:82 in 9998618. [](commit_id = 999861850ca88d1221574c90002b116ed1b0b483, deletion_comment = False)

File: src/JSTest/JSProcess.cs [hunk 3]
```diff
@@ -73,16 +89,18 @@ public bool LaunchProcess(TestProcessStartInfo startInfo, Action<object, string>
                     processExitReceived(p);
                 };
 
-                // EqtTrace.Verbose("ProcessHelper: Starting process '{0}' with command line '{1}'", processPath, arguments);
+                EqtTrace.Verbose("JSProcess: Starting process '{0}' with command line '{1}'", startInfo.FileName, startInfo.Arguments);
```
Comment (cltshivash) [inline src/JSTest/JSProcess.cs]: >EqtTrace.Verbose( [](start = 16, length = 17)

shoukd this go to the logger..?

File: src/JSTest/RuntimeManager.cs [hunk 4]
```diff
@@ -44,6 +43,11 @@ public TestRuntimeManager(JSTestSettings settings, TestRunEvents testRunEvents)
             }
         }
 
+        public int getProcessId()
```
Comment (cltshivash) [inline src/JSTest/RuntimeManager.cs]: >getProcessId [](start = 19, length = 12)

nitpick: fix the casing.. and rename the function to indicate which process..

File: src/JSTest/RuntimeManager.cs [hunk 5]
```diff
@@ -55,56 +59,24 @@ public TestRuntimeManager(JSTestSettings settings, TestRunEvents testRunEvents)
 
         private Action<object, string> ProcessOutputReceived => (process, data) =>
         {
-            Console.Write(data);
-        };
-
-        private Action<object, string> ProcessErrorReceived => (process, data) =>
-        {
-            this.errorReceivedCallback(this.processStdError, data);
+            EqtTrace.Error("JSTestHostManager: Node {0} StdOut: {1}", this.jsProcess.ProcessId, data);
```
Comment (cltshivash) [inline src/JSTest/RuntimeManager.cs]: >Error [](start = 21, length = 5)

is this error ..?

File: src/JSTest/RuntimeManager.cs [hunk 6]
```diff
@@ -55,56 +59,24 @@ public TestRuntimeManager(JSTestSettings settings, TestRunEvents testRunEvents)
...
 
-            var errorString = this.processStdError.ToString();
-            if (!string.IsNullOrEmpty(errorString))
+            if (!string.IsNullOrEmpty(data))
             {
-                //messageLogger.SendMessage(TestMessageLevel.Error, errorString);
-                Console.Write(errorString);
-                // clear waits
+                Console.Write(data + Environment.NewLine);
```
Comment (cltshivash) [inline src/JSTest/RuntimeManager.cs]: >data [](start = 30, length = 4)

suggestion: use string interpolation..

**Comment by cltshivash** [comment]
>                 this.InitializeCommunication(cancellationToken);

Log the time taken here...

---
Refers to: src/JSTest/RuntimeManager.cs:114 in 9998618. [](commit_id = 999861850ca88d1221574c90002b116ed1b0b483, deletion_comment = False)

File: src/JSTest/RuntimeManager.cs [hunk 7]
```diff
@@ -125,17 +97,16 @@ public async Task<bool> LaunchProcessAsync(TestProcessStartInfo runtimeProcessSt
             {
                 try
                 {
-                    this.processStdError = new StringBuilder(this.ErrorLength, this.ErrorLength);
                     EqtTrace.Verbose("JSTestHostManager: Starting process '{0}' with command line '{1}'", runtimeProcessStartInfo.FileName, runtimeProcessStartInfo.Arguments);
 
                     cancellationToken.ThrowIfCancellationRequested();
 
-                    this.jsProcess.LaunchProcess(runtimeProcessStartInfo, this.ProcessErrorReceived, this.ProcessExitReceived);
+                    this.jsProcess.LaunchProcess(runtimeProcessStartInfo, this.ProcessOutputReceived, this.ProcessErrorReceived, this.ProcessExitReceived);
```
Comment (cltshivash) [inline src/JSTest/RuntimeManager.cs]: > this.jsProcess.LaunchProce [](start = 19, length = 27)

log the operation time here as well..

File: src/JSTest.Runner/Environment/Node/DebugLogger.ts [hunk 8]
```diff
@@ -16,42 +20,48 @@ export class DebugLogger implements IDebugLogger {
...
+                    diagFilePath = path.join(diagFile, this.defaultLogFileName() + this.defaultLogFileExtension());
                 }
             } catch (e) {
-                diagFilePath = this.defaultPath();
+                try {
+                    const dirName = path.dirname(diagFile);
+                    const stat = fs.lstatSync(dirName);
+                    if (stat.isDirectory()) {
+                        diagFilePath = path.join(dirName, this.defaultLogFileName() + '-' + path.basename(diagFile));
+                    }
```
Comment (cltshivash) [inline src/JSTest.Runner/Environment/Node/DebugLogger.ts]: can we avoid this code.. and ensure we create a new file each time..?

File: src/JSTest.Runner/Environment/Node/DebugLogger.ts [hunk 9]
```diff
@@ -60,6 +70,10 @@ export class DebugLogger implements IDebugLogger {
     }
 
     public log(message: string) {
-        this.logFileStream.write(message + '\n');
+        if (this.logFileCreated) {
+            this.logFileStream.write(message + '\n');
+        } else {
+            this.messageBuffer.push(message);
```
Comment (cltshivash) [inline src/JSTest.Runner/Environment/Node/DebugLogger.ts]: >messageBuffer [](start = 17, length = 13)

when does the message buffer get flushed..?

File: src/JSTest.Runner/Environment/Node/DebugLogger.ts (hunk 8 above)
Comment (karanjitsingh) [reply src/JSTest.Runner/Environment/Node/DebugLogger.ts]: `this.logFileStream = fs.createWriteStream(diagFilePath);` ensures that, this logic is to check whether the option contained a directory or a file, if it is a file say of file name D:\a\b.log, the file that's created is D:\a\<stamp>-b.log

File: src/JSTest.Runner/Environment/Node/DebugLogger.ts (hunk 9 above)
Comment (karanjitsingh) [reply src/JSTest.Runner/Environment/Node/DebugLogger.ts]: message buffer is for scenarios when the file hasn't been created but logs are being pumped in

```
         this.logFileStream.on('open', () => {
             this.messageBuffer.forEach(message => {
                 this.logFileStream.write(message + '\n');
             });
             this.messageBuffer = [];
             this.logFileCreated = true;
         });
```

in DebugLogger.ts:72 ensures that

File: src/JSTest/JSProcess.cs (hunk 1 above)
Comment (karanjitsingh) [reply src/JSTest/JSProcess.cs]: that way i would also have to de-assign it when the process exits.

i didn't get what you meant by `process [](start = 32, length = 7)`?

File: src/JSTest/JSProcess.cs (hunk 2 above)
Comment (karanjitsingh) [reply src/JSTest/JSProcess.cs]: fixed

File: src/JSTest/RuntimeManager.cs (hunk 5 above)
Comment (karanjitsingh) [reply src/JSTest/RuntimeManager.cs]: fixed

File: src/JSTest/RuntimeManager.cs (hunk 4 above)
Comment (karanjitsingh) [reply src/JSTest/RuntimeManager.cs]: fixed

File: src/JSTest/JSProcess.cs (hunk 1 above)
Comment (karanjitsingh) [reply src/JSTest/JSProcess.cs]: fixed

File: src/JSTest/RuntimeManager.cs [hunk 10]
```diff
@@ -55,56 +61,24 @@ public TestRuntimeManager(JSTestSettings settings, TestRunEvents testRunEvents)
...
-                {
-                    int required = data.Length + testRunnerProcessStdError.Length - testRunnerProcessStdError.MaxCapacity;
-                    if (required > 0)
-                    {
-                        testRunnerProcessStdError.Remove(0, required);
-                    }
-                }
-
-                testRunnerProcessStdError.Append(data);
+                Console.Write(data + Environment.NewLine);
```
Comment (cltshivash) [inline src/JSTest/RuntimeManager.cs]: >Console.Write(data + Environment.NewLine); [](start = 16, length = 42)

prefix stdout and stderr.

File: src/JSTest/JSProcess.cs [hunk 11]
```diff
@@ -45,44 +59,56 @@ public void EnableDebugLogs(string debugFilePath)
...
                     // Call WaitForExit without again to ensure all streams are flushed,
-                    var p = sender as Process;
+                    var exitingProcess = sender as Process;
                     try
                     {
                         // Add timeout to avoid indefinite waiting on child process exit.
-                        p.WaitForExit(500);
+                        if (exitingProcess.WaitForExit(500))
+                        {
+                            EqtTrace.Verbose("JSProcess: Process with id {0} exited successfully.", jsProcessId);
```
Comment (cltshivash) [inline src/JSTest/JSProcess.cs]: >EqtTrace.Verbose [](start = 28, length = 16)

write to the console...

File: src/JSTest/JSProcess.cs [hunk 12]
```diff
@@ -45,44 +59,56 @@ public void EnableDebugLogs(string debugFilePath)
...
+                        }
+                        else
+                        {
+                            EqtTrace.Error("JSProcess: WaitForExit timed out for process {0}", jsProcessId);
```
Comment (cltshivash) [inline src/JSTest/JSProcess.cs]: write to the console..

File: src/JSTest/JSProcess.cs [hunk 13]
```diff
@@ -45,44 +59,56 @@ public void EnableDebugLogs(string debugFilePath)
...
+                        }
                     }
                     catch (InvalidOperationException)
                     {
+                        // Process had already exited
```
Comment (cltshivash) [inline src/JSTest/JSProcess.cs]: >// Process had already exited [](start = 24, length = 29)

write to the console..

File: src/JSTest.Runner/Utils/TimeUtils.ts [hunk 14]
```diff
@@ -1,30 +1,48 @@
...
+    let m: string | number = date.getMinutes();
+    let s: string | number = date.getSeconds();
+
+    h = h < 10 ? '0' + h : h; 
+    m = m < 10 ? '0' + m : m; 
+    s = s < 10 ? '0' + s : s; 
+
+    return [
+        [date.getFullYear(), date.getMonth() + 1, date.getDate() + 1].join(dateJoin),
+        [h, m, date.getSeconds(), date.getMilliseconds()].join(timeJoin)
```
Comment (cltshivash) [inline src/JSTest.Runner/Utils/TimeUtils.ts]: >[h, m, date.getSeconds(), date.getMilliseconds()] [](start = 8, length = 49)

this should be s and not getseconds

**Pull request merged**